src/BlockDiagonals.jl

```diff
@@ -205,28 +205,31 @@ function Base.:*(B1::BlockDiagonal, B2::BlockDiagonal)
     end
 end
 
-function Base.:*(B::BlockDiagonal, M::AbstractMatrix)::Matrix
-    if size(B, 2) !== size(M, 1)
-        throw(DimensionMismatch("A has dimensions $(size(B)) but B has dimensions $(size(M))"))
-    end
+function _check_matmul_dims(A::AbstractMatrix, B::AbstractMatrix)
+    # match error message from LinearAlgebra
+    size(A, 2) == size(B, 1) || throw(DimensionMismatch(
+        "A has dimensions $(size(A)) but B has dimensions $(size(B))"
+    ))
+end
+
+function Base.:*(B::BlockDiagonal, M::AbstractMatrix)
+    _check_matmul_dims(B, M)
     st = 1  # start
     ed = 0  # end
-    d = []
+    d = Matrix[]
```

Comment:
Not much better

Reply:
haha - happy to change it! I just need to re-learn the `map ... do` syntax :)